FarihaIS/VanRoomies
# feat(recommendations): Backend recommendations

# Welcome to VanRoomies! 👋✈️

Fixes: #34 

## Description of the change:
This PR aims to introduce our score-based recommendations for users and listings. It introduces the following two endpoints:

- `GET localhost:3000/api/users/:userId/recommendations/users`
- `GET localhost:3000/api/users/:userId/recommendations/listings`

The reommendations are done on the basis of preference matching as well as via max-min normalization so that each preference contributes a maximum of 1 point to the score of that record. 

## How to manually test:
1. *Run `npm run dev`*
2. Create a few test users, corresponding preferences for them. Add a few listings under the ID of some of the users. For now you can use the Postman collection that does not mandate authentication. It is just important to be able to populate the database a bit. If needed, reach out to myself (@dlalaj) for a copy of my MongoDB collections that you can use for testing.
3. Post a `GET` in each of the two routes above via a userID. Note that the userID should not be for a user who has created listings. You can browse the results in the response to get an idea of how the matches are sorted based on best-overall fit.
